packages/puppeteer-core/src/common/Connection.ts

```diff
@@ -132,19 +131,19 @@ export class CallbackRegistry {
     } catch (error) {
       // We still throw sync errors synchronously and clean up the scheduled
       // callback.
-      callback.promise
-        .valueOrThrow()
-        .catch(debugError)
-        .finally(() => {
-          this.#callbacks.delete(callback.id);
-        });
       callback.reject(error as Error);
-      throw error;
```

Comment:
AFAIK we still need it to properly throw sync errors

Reply:
The `callback.promise.valueOrThrow` will immediately throw because the error rejects.

Reply:
Let's separate the changes to Connection from the chomium-bidi update? so that we can see the test results clearly